src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteEmbeddedLanguage/Infrastructure/EmbeddedSeparatedSyntaxNodeList.cs

```diff
@@ -0,0 +1,105 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.AspNetCore.Analyzers.RouteEmbeddedLanguage.Infrastructure;
+
+internal readonly struct EmbeddedSeparatedSyntaxNodeList<TSyntaxKind, TSyntaxNode, TDerivedNode>
```

Comment:
man... this really makes me want to extract out this part of roslyn into a resuable lib.  or at least allow source embedding...

Reply:
nit: could you add Roslyn as a submodule, and include these files as linked sources?  i'd be happy to do whatever we need in them to ensure they can compile without other dependencies.

Reply:
We're trying to avoid submodules in aspnetcore. I don't have a problem with copy+paste of these.